src/cluster/src/install.rs

```diff
@@ -555,6 +624,36 @@ impl ClusterInstaller {
         Ok(())
     }
 
+    /// Checks if all of the prerequisites for installing Fluvio are met
+    ///
+    /// This will attempt to automatically fix any missing prerequisites,
+    /// depending on the installer configuration. See the following options
+    /// for more details:
+    ///
+    /// - [`with_system_chart`]
+    /// - [`with_update_context`]
+    ///
+    /// [`with_system_chart`]: ./struct.ClusterInstaller.html#method.with_system_chart
+    /// [`with_update_context`]: ./struct.ClusterInstaller.html#method.with_update_context
+    fn check_and_fix(&self) -> Result<(), ClusterError> {
+        // Preforms pre-install checks and tries to fix things if possible
+        match self.pre_install_check() {
```

Comment:
Moving pre-check logic into installer will make Installer more complex.

Reply:
Where do you imagine it all living? I feel like this crate is definitely the place for it. I would consider the `check` module but the `ClusterInstaller` has all of the parameters that you want the check to be compatible with, so for that reason it also kind of makes sense to put it here.

Reply:
see previous comments.  Can make new struct or module, ResourceDependency.  It will be responsible for checking resources exists and return it if needed it.  That way installer can be simpler

Reply:
Is that something that we need to do right now? Can we make an issue for that if we're going to start to doing such a big refactor? I don't really think that's super necessary to do immediately

Reply:
This PR is major API changes. so can defer this and make smallest change now?

Reply:
Since we want to start publishing crates soon, changes should be limited to additional or add minor behavior changes

Reply:
I kind of feel the opposite way actually. I feel like if there are major changes to be made they should be made before the release when they would not count as a breaking change, rather than after the release when they _would_ be a breaking change

Reply:
major changes takes time to discuss review.   the change doesn't seem to be must have for this release